core/dbt/task/show.py

```diff
@@ -0,0 +1,81 @@
+import io
+import threading
+
+from dbt.contracts.results import RunResult, RunStatus
+from dbt.events.functions import fire_event
+from dbt.events.types import ShowNode
+from dbt.exceptions import DbtRuntimeError
+from dbt.task.compile import CompileTask, CompileRunner
+
+
+class ShowRunner(CompileRunner):
+    def __init__(self, config, adapter, node, node_index, num_nodes):
+        super().__init__(config, adapter, node, node_index, num_nodes)
+        self.run_ephemeral_models = True
+
+    def execute(self, compiled_node, manifest):
+        adapter_response, execute_result = self.adapter.execute(
+            compiled_node.compiled_code, fetch=True
+        )
+
+        return RunResult(
+            node=compiled_node,
+            status=RunStatus.Success,
+            timing=[],
+            thread_id=threading.current_thread().name,
+            execution_time=0,
+            message=None,
+            adapter_response=adapter_response.to_dict(),
+            agate_table=execute_result,
```

Comment:
Is this (`agate_table`) how we'd recommend that programmatic invocations access the result set of the `show` command?
```python
>>> from dbt.cli.main import dbtRunner
>>> dbt = dbtRunner()
>>> results, success = dbt.invoke(['show', '--select', 'my_sql_model', '--output', 'json'])
>>> results[0].agate_table.print_table()
| id |
| -- |
|  1 |
# do some io.StringIO() hacking to get JSON value
```

If someone has requested `--output json`, I think the `show` task should also return the JSON-formatted output to the caller, one way or another.

(I understand that the result will also be available in the logs, but this feels like something the result object should really contain directly!)